integration/volume/mount_test.go

```diff
@@ -112,6 +118,113 @@ func TestRunMountVolumeSubdir(t *testing.T) {
 	}
 }
 
+func TestRunMountImage(t *testing.T) {
+	skip.If(t, versions.LessThan(testEnv.DaemonAPIVersion(), "1.48"), "skip test from new feature")
+	skip.If(t, testEnv.DaemonInfo.OSType == "windows", "image mounts not supported on Windows")
+
+	ctx := setupTest(t)
+	apiClient := testEnv.APIClient()
+
+	for _, tc := range []struct {
+		name         string
+		opts         mount.ImageOptions
+		cmd          []string
+		createErr    string
+		startErr     string
+		expected     string
+		skipPlatform string
+	}{
+		{name: "image", cmd: []string{"cat", "/image/foo"}, expected: "bar"},
+		{name: "image_tag", cmd: []string{"cat", "/image/foo"}, expected: "bar"},
+
+		{name: "subdir", opts: mount.ImageOptions{Subpath: "subdir"}, cmd: []string{"ls", "/image"}, expected: "hello"},
+		// {name: "subdir link", opts: mount.ImageOptions{Subpath: "hack/good"}, cmd: []string{"ls", "/image"}, expected: "hello.txt"},
```

Comment:
@LaurentGoderre How about this one?
I assume this is the symlink test?

Reply:
@cpuguy83 Done!